vscode-lean4/src/leanclient.ts

```diff
@@ -100,17 +102,35 @@ export class LeanClient implements Disposable {
     }
 
     async showRestartMessage(): Promise<void> {
-        const restartItem = 'Restart Lean Language Server';
-        const item = await window.showErrorMessage('Lean Language Server has stopped unexpectedly.', restartItem)
-        if (item === restartItem) {
-            void this.start();
+        if (!this.showingRestartMessage) {
+            this.showingRestartMessage = true;
+            const restartItem = 'Restart Lean Server';
+            const item = await window.showErrorMessage('Lean Server has stopped unexpectedly.', restartItem)
+            this.showingRestartMessage = false;
+            if (item === restartItem) {
+                void this.start();
+            }
+        }
+    }
+
+    async showRestartFileMessage(): Promise<void> {
+        if (!this.showingRestartFileMessage) {
```

Comment:
I think you can unify 'showingRestartMessage' and 'showingRestartMessage' as we will never be showing both messages at the same time.